### DISCUSSION TITLE
Support SBOM format in actions

### DISCUSSION THREAD

**Discussion by davidkarlsen** [community/community#18918] [Code Security]
It would be good if the tooling could support SBOM formats: https://anchore.com/sbom/key-things-to-know-about-sboms-and-sbom-standards/
That way you can rely on a standard for software bill of material, rather than having to have implementations for the various languages (like pom.xml etc), sbom tooling already support things like python, maven etc, and thus a bridge for sbom -> github dependency api is all that's needed

https://github.com/CycloneDX/cyclonedx-maven-plugin
https://github.com/anchore/syft

**Comment by stevespringett** [comment]
Support for the CycloneDX standard would be ideal as it’s already [supported by the majority of SCA and related security vendors ](https://cyclonedx.org/about/supporters/), and has already been [standardized upon by GitLab](https://docs.gitlab.com/ee/user/application_security/dependency_scanning/).

Support in Actions would provide an easy path forward for many organizations, and provide the Interop they require.

**Reply by davidkarlsen** [reply]
We started a prototype on https://github.com/evryfs/sbom-dependency-submission-action (not working as of now)

**Comment by harringj** [comment]
SPDX is another important, and widely used, open-source standard. NTIA has come up with a mapping between the standards, so it may be possible to support both: https://www.ntia.gov/files/ntia/publications/sbom_formats_survey-version-2021.pdf.
